tools/deploy-cli/templates/ironic-kustomize.tpl

```diff
@@ -0,0 +1,80 @@
+apiVersion: kustomize.config.k8s.io/v1beta1
+kind: Kustomization
+resources:
+- https://github.com/metal3-io/baremetal-operator/config/namespace/
+{{ if and .DeployBasicAuth .DeployTLS }}
+- https://github.com/metal3-io/baremetal-operator/ironic-deployment/overlays/basic-auth_tls/
+{{ else if .DeployBasicAuth }}
+- https://github.com/metal3-io/baremetal-operator/ironic-deployment/base/
+{{ end }}
+namePrefix: baremetal-operator-
+namespace: baremetal-operator-system
+components:
+{{ if and .DeployBasicAuth (not .DeployTLS) }}
+- https://github.com/metal3-io/baremetal-operator/ironic-deployment/components/basic-auth/
+{{ end }}
+{{ if .DeployKeepAlived }}
+- https://github.com/metal3-io/baremetal-operator/ironic-deployment/components/keepalived/
+{{ end }}
+{{ if .DeployMariadb }}
+- https://github.com/metal3-io/baremetal-operator/ironic-deployment/components/mariadb/
+{{ end }}
+
+{{ if .DeployBasicAuth }}
+secretGenerator:
+- envs:
+  - ironic-htpasswd
+  name: ironic-htpasswd
+  type: Opaque
+{{ end }}
+configMapGenerator:
+{{/* The keepalived component has its own configmap, but we are overriding depending on environment here so we must replace it.*/}}
+{{ if .DeployKeepAlived }}
+- behavior: replace
+{{ else }}
+- behavior: create
+{{ end }}
+  envs:
+  - ironic_bmo_configmap.env
+  name: ironic-bmo-configmap
+{{/* This configMap is solely so that we can replace the IP placeholders */}}
```

Comment:
If we have the value in the ironic_bmo_configmap.env we can use that instead. This is fine like it is though, just wanted to mention it in case we can simplify.

Reply:
Not sure I understood your point. Could you please elaborate?

Reply:
Thank you! It is more elegant that way. I was just having an impression that the config map values can only come from either a file or literals, not both. Happy to be wrong though xD

Reply:
It is possible to combine all in one. You can have a configmap with all three `envs`,  `files` and `litterals` :slightly_smiling_face: